pytorch_lightning/plugins/environments/slurm_environment.py

```diff
@@ -69,8 +66,19 @@ def master_port(self) -> int:
 
         return int(default_port)
 
-    def world_size(self):
-        return None
+    def world_size(self) -> int:
+        return int(os.environ["SLURM_NTASKS"])
+
+    def set_world_size(self, size: int) -> None:
+        # no-op, we are not allowed to change world size in SLURM
+        pass
+
+    def global_rank(self) -> int:
+        return int(os.environ["SLURM_PROCID"])
+
+    def set_global_rank(self, rank: int) -> None:
+        # no-op, we are not allowed to change rank in SLURM
+        pass
```

Comment:
Maybe better raise an Error here to make that more explicit?

Because if I call a function like this and it doesn't error I expect the new rank to be set.

Reply:
hmm, but then we would have to make the call to the setter conditional in the training plugins. 
For example, in DDPSpawnPlugin we use the setter to update the global rank once we have spawned the new processes.

Reply:
cc @ananthsub

Reply:
Maybe we can make add log an error here to warn users but not raise an exception?

Reply:
Just a warning would be fine. Might be a good time to introduce several warn / log level depending on risks.

Reply:
I went for log.debug for now. I don't want to risk users seeing error messages pop up that they don't understand. The team wants to include this PR in in the patch release today. I could not find a smarter way to avoid the setter.